File: src/challenge273_easy/Solution.java
package challenge273_easy;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

/**
 * Created by Jonathan on 7/16/16.
 */
public class Solution {

    private static final String DEGREE_UNIT = "d";
    private static final String RADIAN_UNIT = "r";

    public static void main (String[] args) {

        Scanner scanner = new Scanner(System.in);
        System.out.println("Please enter input filename: ");
        String inputFileName = scanner.nextLine();

        Pattern numberValueRegex = Pattern.compile("[0-9]+.?[0-9]+");
        Pattern unitRegex = Pattern.compile("[a-z]{2}$");

        try(BufferedReader bufferedReader = new BufferedReader(new FileReader("input-files/challenge273/" + inputFileName))) {
            String nextLine = bufferedReader.readLine();
            while (nextLine != null) {
                Matcher numberValueMatcher = numberValueRegex.matcher(nextLine);
                numberValueMatcher.find();
                Float numberValue = Float.parseFloat(numberValueMatcher.group(0));

                Matcher unitMatcher = unitRegex.matcher(nextLine);
                unitMatcher.find();
                String unit = unitMatcher.group(0);
                String fromUnit = Character.toString(unit.charAt(0));
                String toUnit = Character.toString(unit.charAt(1));

                if (RADIAN_UNIT.equals(fromUnit) && DEGREE_UNIT.equals(toUnit)) {
                    System.out.println(radianToDegrees(numberValue) + DEGREE_UNIT);
                } else if (DEGREE_UNIT.equals(fromUnit) && RADIAN_UNIT.equals(toUnit)) {
                    System.out.println(degreesToRadians(numberValue) + RADIAN_UNIT);
                } else {
                    System.out.println("Invalid input entered.");
                }
                nextLine = bufferedReader.readLine();
            }
        } catch (Exception exception) {
            System.out.println(exception.toString());
        }
    }

    private static double radianToDegrees(double numOfRadians) {
        return (numOfRadians * (180 / Math.PI));
    }

    private static double degreesToRadians(double numOfDegrees) {
        return (numOfDegrees * (Math.PI / 180));
    }
}

File: src/challenge273_easy/Solution.py
import re, math

def radians_to_degrees(radians):
	return radians * (180/math.pi)

def degrees_to_radians(degrees):
	return degrees * (math.pi/180)

valueRegex = re.compile("[0-9]+.?[0-9]+")
unitRegex = re.compile("[a-z]{2}$")


with open("../../input-files/challenge273/inputFile.txt", "r") as inputFile:
	for line in inputFile:
		valueMatch = valueRegex.search(line)
		value_to_convert = valueMatch.group()

		unitMatch = unitRegex.search(line)
		units_list = list(unitMatch.group())

		if units_list[0] == "d" and units_list[1] == "r":
			print str(degrees_to_radians(float(value_to_convert))) + "r"
		elif units_list[0] == "r" and units_list[1] == "d":
			print str(radians_to_degrees(float(value_to_convert))) + "d"
		else:
			print "Invalid input entered."

